### DISCUSSION TITLE
Can I join 2 datasets via SQL from different underlying databases

### DISCUSSION THREAD

**Discussion by majkinetor** [apache/superset#22437] [Q&A / Help]
Say I have sql server table and CSV table as 2 datasets. 

I want to join 2 tables.

I see on [MR](https://github.com/apache/superset/pull/15241) this comment by @rumbin which casts doubt:

> Especially when two datasets are to be joined this way, they need to refer to the same database.  

I dont see a reason why this would be the case in general. 

My case is this: I have one table wich shows entity code, and I want entity name, however, that database doesn't contain that info and I must import it via CSV that contains entity code and name. I want then to join 2 tables and show entity name instead of code. That requuires joining 1 dataset coming from Sql Server and 1 coming from CSV (redash supports this for example).

**Comment by rumbin** [comment]
Superset's design philosophy is to delegate the heavy lifting of filtering, joining and aggregating to a database. Thus, all data that has to be processed in a single query needs to reside on a single database, or to be more precise, it needs to be accessible via a single database connection.
The latter may also be a big data query engine (Presto, Dremio,...) which, in turn, can push queries to multiple source DBs at once and join the results.
Your use case of enriching existing data with some dimensions provided by a CSV file can only be solved by uploading the CSV to the database (connection) where the other data that you want to query resides.
This can even be achieved with Superset's CSV upload functionality.
But, imho, the cleaner approach is to integrate the dimension information into the analytic DB via well maintained and documented data pipelines, in order to ensure that the data is up to date and complete.

**Reply by majkinetor** [reply]
All that fail if you don't control the database in question. And providing analytical db for myself only to have this is tremendous job,  that could be achieved naturaly.

Redash is great example of how this works flowlessly and with 0 effort, see [query results data source](https://redash.io/help/user-guide/querying/query-results-data-source/) for details. It keeps query results in the internal sqlite db and then you can do whatever, even joining data with REST json results and other crazy but awesome stuff like that.

**Reply by majkinetor** [reply]
> Superset's design philosophy is to delegate the heavy lifting of filtering, joining and aggregating to a database.

Even this would still be true. Database still do the work, just another special one. Sqlite seems like fantastic choice here, but one could even extrapolate on that and define custom connection for this.
